Tools/cases_generator/stacking.py

```diff
@@ -443,19 +458,41 @@ def write_components(
                 mgr.instr.write_body(out, -4, mgr.active_caches, tier)
 
         if mgr is managers[-1] and not next_instr_is_set:
-            # TODO: Explain why this adjustment is needed.
+            # Adjust the stack to its final depth, *then* write the
+            # pokes for all preceding uops.
+            # Note that for array output effects we may still write
+            # past the stack top.
```

Comment:
This seems surprising to me. When could this happen?

Reply:
E.g. in `UNPACK_SEQUENCE` and its specializations. For array outputs, the variable is just a pointer, and it is initialized *before* the body of the opcode is executed. The body is supposed to write its output directly into the stack using this pointer. But the SP is not adjusted until after the body is executed. This is iffy only if you believe that something else might write over the portion of the stack above the SP while that code is running. But nothing ever does that.